### ISSUE TITLE
smb Demo

### ISSUE THREAD

**Issue by rmatinata** [lscharen/iigs-game-engine#28]
Just curious... Is the SMB demo from `master` supposed to be working ? At the moment, it compiles, but all it does when launched from GS/OS is to initialize the playfield and then quit back to GS/OS...

Or did I get it wrong ?

**Comment by lscharen** [comment]
Yes, the `master` branch is not working.  If you switch to the `eight-bit-hal` branch that will build Tool160 and the SMB demo in a runnable state.

**Comment by rmatinata** [comment]
Got it to work in GSplus... Thanks !

It does crash in Accelerated (ZipGS) mode, though, right after the initial Video/Audio quality selection screen:

<img width="701" alt="image" src="https://github.com/user-attachments/assets/b6be035f-a701-4509-b490-53312bb3eba9">
